subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -269,6 +340,9 @@ static int pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 		cmd.output.timer_count = buf[pos++];
 		cmd.output.timer_count |= buf[pos++] << 8;
 		ret = OSDP_PD_ERR_REPLY;
+		if (!pd_cmd_cap_ok(pd, &cmd)) {
+			break;
+		}
```

Comment:
Move capablity checking before setting any reply information.
It applies here and for CMD_LED, CMD_BUZ and CMD_TEXT.

Reply:
Not sure I follow. Which reply information are you taking about?

Reply:
I.e. in this case `if (!pd_cmd_cap_ok(pd, &cmd)) {...}` should be moved just after `if (len != CMD_OUT_DATA_LEN || !pd->command_callback) {...}`

Reply:
Ah, that. No we cannot do that; the cap check needs the command that we decoded in som cases (wherever cmd arg is non-null).